test_main.py

```diff
@@ -66,30 +70,131 @@ def test_url_caching(empty_deployment):
+                "origin": "_",
+                "log_url": "https://drive.google.com/u/0/uc?id=1EeWjyLJG3f8K7U2SxwOGYnAvSG01YM4g&export=download",
+            },
+            {
+                "build_id": "kernelci:kernelci.org:64147283e6021132258c86c0",
+                "id": "_:1",
+                "origin": "_",
+                "output_files": [
+                    {
+                        "name": "x86_64_4_console.log",
+                        "url": "https://github.com/kernelci/kcidb/blob/main/setup.py?padding=4673"
+                    },
+                    {
+                        "name": "x86_64_4_IOMMU_boot_test_dmesg.log",
+                        "url": "https://github.com/kernelci/kcidb/blob/main/requirements.txt?padding=1649"
+                    },
+                ],
+                "environment": {
+                    "comment": "meson-s905d-p230 in lab-baylibre",
+                    "misc": {
+                        "rootfs_url": "http://storage.kernelci.org/images/rootfs/buildroot/buildroot-baseline/20230120.0/armel/rootfs.cpio.gz"
+                    }
+                },
+            },
+        ],
+    }
+
+    # Submit data to submission queue
+    client.submit(data)
+
+    current_time = time.time()
+    # The time when caching system should be done with all our data
+    deadline_time = current_time + 300
+
+    # URLs to be check, as they should be cached
     urls_messages = [
-        ["https://github.com/kernelci/kcidb/blob/main/setup.py"],
-        ["https://github.com/kernelci/kcidb/blob/main/requirements.txt",
-         "https://github.com/kernelci/kcidb/blob/main/README.md"]
+        ["https://github.com/kernelci/kcidb/blob/main/setup.py?padding=4673"],
+        ["https://github.com/kernelci/kcidb/blob/main/requirements.txt?padding=1649"],
+        ["https://github.com/kernelci/kcidb/blob/main/.gitignore?padding=547"],
+        ["https://github.com/kernelci/kcidb/blob/main/.pylintrc?padding=247",
+        "https://github.com/kernelci/kcidb/blob/main/doc/_index.md?padding=7230"],
+        ["https://github.com/kernelci/kcidb/blob/main/.github/workflows/deploy.yml?padding=4821"],
+        ["https://github.com/kernelci/kcidb/blob/main/doc/installation.md?padding=1505"]
     ]
 
-    for urls in urls_messages:
-        pub_client.publish(urls)
-
     # Retry checking URLs in the cache for a minute
     retry_interval = 5  # seconds
-    max_retries = 12  # 60 seconds / 5 seconds
 
     for urls in urls_messages:
         for url in urls:
-            for _ in range(max_retries):
+            while True:
                 if check_url_in_cache(url):
                     break
+                if time.time() > deadline_time:
+                    raise AssertionError(f"URL '{url}' not found in the cache")
                 time.sleep(retry_interval)
-            else:
-                raise AssertionError(f"URL '{url}' not found in the cache")
+
+    current_time = time.time()
+
```

Comment:
Could you remove this blank line? It doesn't help and makes the code longer.

Reply:
okay